ament_lint_cmake/ament_lint_cmake/cmakelint.py

```diff
@@ -382,6 +386,16 @@ def IsValidFile(filename):
     return filename.endswith('.cmake') or os.path.basename(filename).lower() == 'cmakelists.txt'
 
 def ProcessFile(filename):
+    # Store and then restore the filters to prevent pragmas in the file from persisting.
+    original_filters = list(_lint_state.filters)
+    try:
+        return _ProcessFile(filename)
```

Comment:
Wouldn't this be cleaner to pass in a copy of the filters instead of copying mutating and restoring?